### DISCUSSION TITLE
请教一下DNS流程和嗅探选项的问题

### DISCUSSION THREAD

**Discussion by Rashhz** [Openwrt-Passwall/openwrt-passwall2#98] [Q&A]
1、直连dns设置为udp（119.29.29.29），远程dns设置为fakeDNS时的dns流程：
“本机上的应用（例如浏览器）发到路由器的dns，经过劫持到路由器的53端口，passwall得到dns后经过xray的路由进行dns分流，将包含需要直连的域名的dns请求发到119.29.29.29，将包含需要代理的域名的dns请求返回Fakeip”

我对dns的流程的这个理解是正确的吗？

2.关于嗅探仅供路由使用这个选项。
这个选项为勾选的情况下，passwall将把在dns阶段解析出的ip直接交给远程代理服务器，代理服务器将直接对ip发起连接。
这个选项为不勾选的情况下，passwall所做的dns解析仅做路由使用，将经过路由后需要代理的域名发给远程代理服务器，由代理服务器dns解析后再连接。如果请求是基于ip的，则嗅探出域名后，经过路由，再把域名交给远程服务器。我的这个理解是正确的吗？

感谢大佬

**Comment by Rashhz** [comment]
还有就是，passwall2是如何得到dns请求的？通过iptables重定向53端口到passwall吗？

**Reply by a904055262** [reply]
你机器的网关和DNS都是路由器ip。。。
